firebase-firestore/src/main/java/com/google/firebase/firestore/Query.java

```diff
@@ -900,6 +899,47 @@ private ListenerRegistration addSnapshotListenerInternal(
         firestore.getClient(), queryListener, activity, wrappedListener);
   }
 
+  @PublicApi
+  public String getPath() {
+    return query.getPath().canonicalString();
+  }
+
+  @PublicApi
+  public List<com.google.firebase.firestore.Filter> getFilters() {
+    com.google.firebase.firestore.Filter newFilter;
+    List<com.google.firebase.firestore.Filter> filters = new ArrayList<>();
+    for (Filter filter : query.getFilters()) {
+      newFilter = new com.google.firebase.firestore.Filter(filter);
+      filters.add(newFilter);
+    }
+    return filters;
+  }
+
+  @PublicApi
+  public List<OrderBy> getExplicitOrderBy() {
+    return query.getExplicitOrderBy();
+  }
+
+  @PublicApi
+  public boolean hasLimit() {
+    return query.hasLimit();
+  }
+
+  @PublicApi
+  public long getLimit() {
+    return query.getLimit();
+  }
+
+  @PublicApi
+  public @Nullable String getStartAt() {
```

Comment:
I think getStartAt() / getEndAt() need to return `Object`, like `getFieldValue()` on `Filter` does, so that you could pass the value back into `.startAt(...)` to reconstruct the query...  But we also need to be able to distinguish the lack of a startAt from the `null` startAt.  So we may need `.hasStartAt()` that returns bool and `.getStartAt()` that returns the `Object` value of the startAt (and probably throws an exception if there is no startAt).